apps/ensapi/src/lib/ensanalytics/database.ts

```diff
@@ -0,0 +1,112 @@
+import { getUnixTime } from "date-fns";
+import { and, count, desc, eq, gte, isNotNull, lt, ne, sql, sum } from "drizzle-orm";
+import { zeroAddress } from "viem";
+
+import * as schema from "@ensnode/ensnode-schema";
+import {
+  type AccountId,
+  deserializeDuration,
+  serializeAccountId,
+  type UnixTimestamp,
+} from "@ensnode/ensnode-sdk";
+
+import { db } from "@/lib/db";
+import type { AggregatedReferrerSnapshot } from "@/lib/ensanalytics/types";
+import { ireduce } from "@/lib/itertools";
+import logger from "@/lib/logger";
+
+/**
+ * Fetches all referrers with 1 or more qualified referrals from the `registrar_actions` table and builds an `AggregatedReferrerSnapshot`.
```

Comment:
```suggestion
 * Fetches all referrers with 1 or more qualified referrals from the `registrar_actions` table
 * and builds an `AggregatedReferrerSnapshot`.
```

Goal: optimize max line widths